control-plane/pkg/prober/async_prober_test.go

```diff
@@ -158,3 +215,99 @@ func TestAsyncProber(t *testing.T) {
 		})
 	}
 }
+
+// Adapted from https://github.com/knative/eventing/blob/57d78e060db6e0d2f3046eeedd27137cfa4fe0bc/pkg/eventingtls/eventingtlstesting/eventingtlstesting.go#L84
```

Comment:
Did you consider changing the one in Eventing instead adding our own?
I'm sure we will need the same feature for other use cases

Reply:
I didn't think of this at the time because the one in Eventing isn't exported, but I can see why it would be a good idea to add this there. I'll open a PR